### DISCUSSION TITLE
Make it easier to bookmark password protected shared links

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#419] [Anything Else]
"If you bookmark the page you get to after typing the password, you'll get an error message saying you don't have access (you have to bookmark the page where you enter your password, which might not make sense to some people – and it doesn't look like the browser can save the password)"

(feedback via email)

**Comment by Vigasaurus** [comment]
I'm not sure what the resolution to this would look like. If we redirected all sites with password protected links to the password screen, that probably makes it too easy for people to get to the password screen without knowing the "impossible to guess" link. And if we leave it as it is, the issues brought up exist. 

I guess the biggest resolution will just end up being user accounts - with viewer only accounts or similar.

**Reply by ukutaht** [reply]
I have an idea but I haven't tested it yet.

Instead of the shared link being `plausible.io/share/<slug>` it could be `plausible.io/<domain>?auth=<slug>` or something like this. This way we dont need to redirect the person from `plausible.io/share/<slug>` -> `plausible.io/<domain>`. Without the redirection, the person should be able to bookmark the  `plausible.io/<domain>?auth=<slug>` link and they would just have to re-enter the password when the authorization expires.

writing this I realized that slug is probably a bad name for the ID of the secret link...

**Reply by Vigasaurus** [reply]
Yeah I quite like that - I think that also can help to make it easy for those who get access to multiple shared links to know which is which without having to open them all up

**Comment by metmarkosaric** [comment]
Our private and secure shared links now have a new format which makes them easier for your clients to bookmark and access again. If password protected, the password is now saved too so the client doesn't need to type the password on each and every visit. Don't worry, any existing shared link will automatically redirect to the new one.

**Reply by robertvanhoesel** [reply]
A small security improvement and neat trick I saw being used by Excalidraw is saving the auth key in the hash fragment of the url, rather than the query fragment. The app would then read the hash client side and use it to authenticate. This way, the auth key is never send plain in the url but can still be bookmarked. I.e. `plausible.io/<domain>#auth=<slug>`

**Reply by ukutaht** [reply]
Good point, thanks for the suggestion!
